src/main/java/com/aws/greengrass/logmanager/model/LogFileGroup.java

```diff
@@ -0,0 +1,72 @@
+package com.aws.greengrass.logmanager.model;
+
+import com.aws.greengrass.util.Utils;
+import lombok.Getter;
+
+import java.io.File;
+import java.net.URI;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import static com.aws.greengrass.logmanager.LogManagerService.ACTIVE_LOG_FILE_FEATURE_ENABLED_FLAG;
+
+public final class LogFileGroup {
+    @Getter
+    List<LogFile> logFiles;
```

Comment:
Not good don't make it public enforce that callers have to tell this Group what to do on each of the stored files. This breaks object orientation https://martinfowler.com/bliki/TellDontAsk.html